src/models/dispatcher/index.ts

```diff
@@ -1,23 +1,48 @@
 /* eslint-disable @typescript-eslint/ban-types */
 
-import type IDispatcher from "@/interfaces/idispatcher.js";
+import type { default as DispatcherInterface } from "@/interfaces/idispatcher.js";
+import Resolver from "@/interfaces/iresolver";
+import ResolverImplementation from "@/models/resolver.js"
 import { BehaviorMappings, NotificationMappings } from "@/models/dispatcher/mappings.js";
 
+export type OrderMapping<TData = {}> = TData & {
+    order?: number;
+}
+
 /**
  * The internal dispatcher
  * Here storing the mapping between the event name with the handler name
  * 
  * @export
  * @class Dispatcher
- * @implements {IDispatcher}
+ * @implements {Dispatcher}
  */
-export default class Dispatcher implements IDispatcher {
+export default class Dispatcher implements DispatcherInterface {
+    private static _instance: DispatcherInterface | undefined;
+
     private readonly _notifications: NotificationMappings;
     private readonly _behaviors: BehaviorMappings;
 
-    constructor() {
-        this._notifications = new NotificationMappings();
-        this._behaviors = new BehaviorMappings();
+    public static get instance() {
+        if(this._instance) {
+            return this._instance;
+        }
+
+        const instance = new Dispatcher(ResolverImplementation.instance);
+        this._instance = instance;
+
+        return instance;
```

Comment:
Sure - the reason I typically use local variables for this is so that the ordering of the lines doesn't matter. That way, the compiler will tell you if the order changed.

But since this is one line, that is overkill, and I'll change it 👍